concurrency/MpmcQueue.h

```diff
@@ -77,7 +77,6 @@ class MpmcQueue
                 // the write cannot reach here until the state is set empty.
                 auto& entry = _elements[pos % _maxElements];
                 target = std::move(entry.value);
-                entry.value.~T();
```

Comment:
Good thing 👍 . This seems a very danger thing to do on in auto initialised memory.
Maybe it would be even better change `entry.value` to an uninitialized memory like `std::aligned_storage<T>`  and call `T()` and `~T` explicitly.